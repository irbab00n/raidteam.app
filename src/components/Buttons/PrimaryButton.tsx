import styled from 'styled-components';

interface PrimaryButtonProps {
  width: string;
  height: string;
  borderColor: string;
  borderRadius: string;
  borderSize: string;
  borderStyle: string;
  backgroundColor: string;
  clickBackgroundColor: string;
  hoverBackgroundColor: string;
  fontColor: string;
  fontSize: string;
  xPadding: string;
  yPadding: string;
}

const PrimaryButton = styled.button`
  transition: 0.2s;
  cursor: pointer;
  height: ${(props: PrimaryButtonProps) =>
    props.height ? props.height : '40px'};
  min-width: ${(props: PrimaryButtonProps) =>
    props.width ? props.height : '100px'};

  background-color: ${(props: PrimaryButtonProps) =>
    props.backgroundColor ? props.backgroundColor : 'rgba(159, 22, 0, 0.6)'};

  color: ${(props: PrimaryButtonProps) =>
    props.fontColor ? props.fontColor : '#f8b700'};
  font-size: ${(props: PrimaryButtonProps) =>
    props.fontSize ? props.fontSize : '16px'};

  border-size: ${(props: PrimaryButtonProps) =>
    props.borderSize ? props.borderSize : '2px'};
  border-radius: ${(props: PrimaryButtonProps) =>
    props.borderRadius ? props.borderRadius : '4px'};
  border-style: ${(props: PrimaryButtonProps) =>
    props.borderStyle ? props.borderStyle : 'solid'};
  border-color: ${(props: PrimaryButtonProps) =>
    props.borderColor ? props.borderColor : 'rgb(234, 48, 18)'};

  padding-left: ${(props: PrimaryButtonProps) =>
    props.xPadding ? props.xPadding : '10px'};
  padding-right: ${(props: PrimaryButtonProps) =>
    props.xPadding ? props.xPadding : '10px'};
  padding-bottom: ${(props: PrimaryButtonProps) =>
    props.yPadding ? props.yPadding : '0px'};
  padding-top: ${(props: PrimaryButtonProps) =>
    props.yPadding ? props.yPadding : '0px'};

  &:hover {
    background-color: ${(props: PrimaryButtonProps) =>
      props.hoverBackgroundColor ? props.hoverBackgroundColor : '#9f1600'};
  }

  &:active {
    background-color: ${(props: PrimaryButtonProps) =>
      props.clickBackgroundColor ? props.clickBackgroundColor : '#660e00'};
    color: #ffd36b;
    outline: 0;
  }

  &:focus {
    outline-width: 0;
  }
`;

export default PrimaryButton;
