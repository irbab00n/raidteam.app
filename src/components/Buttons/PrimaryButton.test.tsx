import * as React from 'react';
import * as sinon from 'sinon';
import { shallow, mount } from 'enzyme';

import PrimaryButton from './PrimaryButton';

describe('PrimaryButton', () => {
  it('Should render correctly with no props', () => {
    const component = shallow(<PrimaryButton />);
    expect(component).toMatchSnapshot();
  });

  it('Allows props to be set', () => {
    const testWidth = '250px';
    const testHeight = '40px';
    const component = mount(
      <PrimaryButton width={testWidth} height={testHeight} />
    );
    expect(component.props().width).toEqual(testWidth);
    expect(component.props().height).toEqual(testHeight);
  });

  it('Allows children to be passed through', () => {
    const child = <div className="button-label">Label</div>;
    const component = mount(<PrimaryButton>{child}</PrimaryButton>);
    expect(component.contains(child)).toEqual(true);
    expect(component.contains('some other value')).toEqual(false);
  });

  it('Allows external classNames to be supplied', () => {
    const testClassName = 'test-class-name';
    const component = mount(<PrimaryButton className={testClassName} />);
    expect(component.hasClass(testClassName)).toEqual(true);
    expect(component.hasClass('some-failure-class')).toEqual(false);
  });

  it('Allows an onClick function to be supplied and used', () => {
    const onClick = sinon.spy();
    const component = shallow(<PrimaryButton onClick={onClick} />);
    component.simulate('click');
    expect(onClick.calledOnce).toEqual(true);
  });
});
