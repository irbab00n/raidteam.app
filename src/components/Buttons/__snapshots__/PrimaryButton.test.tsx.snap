// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton Should render correctly with no props 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  transition: 0.2s;
  cursor: pointer;
  height: ",
          [Function],
          ";
  min-width: ",
          [Function],
          ";

  background-color: ",
          [Function],
          ";

  color: ",
          [Function],
          ";
  font-size: ",
          [Function],
          ";

  border-size: ",
          [Function],
          ";
  border-radius: ",
          [Function],
          ";
  border-style: ",
          [Function],
          ";
  border-color: ",
          [Function],
          ";

  padding-left: ",
          [Function],
          ";
  padding-right: ",
          [Function],
          ";
  padding-bottom: ",
          [Function],
          ";
  padding-top: ",
          [Function],
          ";

  &:hover {
    background-color: ",
          [Function],
          ";
  }

  &:active {
    background-color: ",
          [Function],
          ";
    color: #ffd36b;
    outline: 0;
  }

  &:focus {
    outline-width: 0;
  }
",
        ],
      },
      "displayName": "styled.button",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "button",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
